sphinx/ext/autodoc/__init__.py

```diff
@@ -1542,15 +1540,14 @@ class AutoDirective(Directive):
 
     def warn(self, msg):
         # type: (unicode) -> None
-        self.warnings.append(self.reporter.warning(msg, line=self.lineno))
+        logger.warning(msg, line=self.lineno)
```

Comment:
I'm assuming this where is the issue lies. Why can't we call warnings and still return this?